src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -774,6 +764,37 @@ CompletableFuture<Void> spawnTerminationListener(Process process, Boolean forOnl
     });
   }
 
+  @VisibleForTesting
+  void spawnReplayTerminationListener(Process process) {
+    process.onExit().thenAccept(finishedProcess -> {
+      handleTermination(finishedProcess);
+      setReplayRunning(false);
+    });
+  }
+
+  private void handleTermination(Process finishedProcess) {
+    fafServerAccessor.setPingIntervalSeconds(25);
+    int exitCode = finishedProcess.exitValue();
+    log.info("Forged Alliance terminated with exit code {}", exitCode);
+    Optional<Path> logFile = loggingService.getMostRecentGameLogFile();
+    logFile.ifPresent(file -> {
+      try {
+        Files.writeString(file, logMasker.maskMessage(Files.readString(file)));
+      } catch (IOException e) {
+        log.warn("Could not open log file", e);
+      }
+    });
+
+    if (exitCode != 0 && !(gameKilled || replayKilled)) {
+      if (exitCode == -1073741515) {
+        notificationService.addImmediateWarnNotification("game.crash.notInitialized");
+      } else {
+        notificationService.addNotification(new ImmediateNotification(i18n.get("errorTitle"), i18n.get("game.crash", exitCode, logFile.map(Path::toString)
+            .orElse("")), WARN, List.of(new Action(i18n.get("game.open.log"), event -> platformService.reveal(logFile.orElse(operatingSystem.getLoggingDirectory()))), new DismissAction(i18n))));
+      }
+    }
```

Comment:
The main thing with breaking apart the gameKilled and replayKilled was so that replays would not interfere with this notification for games that crashed.